lib/xpm_ruby/connection.rb

```diff
@@ -16,9 +16,14 @@ def get(endpoint:, params: nil)
 
       case response.status
       when 401
-        hash = Ox.load(response.body, mode: :hash_no_attrs, symbolize_keys: false)
+        detail = JSON.parse(response.body)["Detail"]
 
-        raise Unauthorized.new(hash["html"]["head"]["title"])
+        case detail
+        when /TokenExpired: token expired/
+          raise AccessTokenExpired.new(detail)
+        else
+          raise Unauthorized.new(detail)
```

Comment:
Just FYI @abreckner I've removed the explicit `AccessTokenRejected` error and will instead just raise the base `Unauthorized` one.

Reason is based on some [discussion with Louie about WorkflowMax errors on the staff endpoint](https://ignitionapp.slack.com/archives/C0125F6R353/p1587624627091600).

In summary I think I was wrong to assume that a 401 is specifically due to access token issue so am just going to bubble it up 401 errors verbatim (except for `AccessTokenExpired` errors which is useful for the caller to distinguish).

Reply:
👍